pmlc/conversion/tile_to_pxa/pass.cc

```diff
@@ -182,6 +183,42 @@ struct ScalarConstantOpConversion : public OpConversionPattern<ew::ScalarConstan
   }
 };
 
+Value createCastOp(ConversionPatternRewriter& rewriter, Location loc, Value from, Type intoType, bool isSigned) {
+  auto fromType = from.getType();
+  if (fromType == intoType) {
+    return from;
+  }
+  if (auto intoFloatType = intoType.dyn_cast<FloatType>()) {
+    if (auto fromFloatType = fromType.dyn_cast<FloatType>()) {
+      if (fromFloatType.getWidth() < intoFloatType.getWidth()) {
+        // FPExtOp: FloatType -> wider FloatType
+        return rewriter.create<mlir::FPExtOp>(loc, from, intoType).getResult();
+      }
+      // FPTruncOp: FloatType -> narrower FloatType
+      return rewriter.create<mlir::FPTruncOp>(loc, from, intoType).getResult();
```

Comment:
Technically there is an uncovered case here of fp16 to bf16, which I think we should ignore, but just as an FYI.

Reply:
Also there's a missing case to go from integer to floating point.